avocado_vt/plugins/vt_runner.py

```diff
@@ -0,0 +1,279 @@
+import logging
+import multiprocessing
+import os
+import sys
+import tempfile
+import time
+import traceback
+
+from avocado.core import exceptions, nrunner, output
+from avocado.utils import stacktrace, path
+
+from avocado_vt import utils
+from virttest import (data_dir, env_process, error_event, utils_env, utils_misc,
+                      utils_params, version, funcatexit)
+
+
+def _send_message(msg, queue, message_type):
+    status = {'type': message_type, 'log': msg}
+    queue.put(status)
+
+
+class RunnerLogHandler(logging.Handler):
+
+    def __init__(self, queue, message_type):
+        """
+        Runner logger which will put every log message to the runner queue
+
+        :param queue: queue for the runner messages
+        :type queue: :class: multiprocessing.SimpleQueue
+        :param message_type: type of the log message
+        :type message_type: string with possible values: {stdout, stderr, log}
+        """
+        super().__init__()
+        self.queue = queue
+        self.message_type = message_type
+
+    def emit(self, record):
+        msg = self.format(record)
+        _send_message(msg, self.queue, self.message_type)
+
+
+class StreamToQueue:
+
+    def __init__(self, queue, message_type):
+        """
+        Runner Stream which will transfer every message to the runner queue
+
+        :param queue: queue for the runner messages
+        :type queue: multiprocessing.SimpleQueue
+        :param message_type: type of the log
+        :type message_type: string
+        """
+        self.queue = queue
+        self.message_type = message_type
+
+    def write(self, buf):
+        _send_message(buf, self.queue, self.message_type)
+
+    def flush(self):
+        pass
+
+
+class VirtTest(utils.TestUtils):
+
+    def __init__(self, queue, runnable):
+        self.__vt_params = utils_params.Params(runnable.kwargs)
+        self.queue = queue
+        self.tmpdir = tempfile.mkdtemp()
+        self.logdir = os.path.join(self.tmpdir, 'results')
+        path.init_dir(self.logdir)
+        self.logfile = os.path.join(self.logdir, 'debug.log')
+        self.log = output.LOG_JOB
+        self.log_level = runnable.config.get('job.output.loglevel',
+                                             logging.DEBUG)
+        self.env_version = utils_env.get_env_version()
+        self.iteration = 0
+        self.background_errors = error_event.error_events_bus
+        # clear existing error events
+        self.background_errors.clear()
+        self.debugdir = self.logdir
```

Comment:
The original VirtTest class also has `resultsdir` but I admit this is too many names for the same thing and will lead to confusion. However, it may also lead to breakage so it is worth to use just logdir or debugdir later on. Regarding the previously reported error:
```
16:03:47 ERROR| 'VirtTest' object has no attribute 'outputdir'
16:03:48 WARNI| Could not get host cpu family
16:03:48 ERROR| 
16:03:48 ERROR| Reproduced traceback from: /usr/lib/python3.9/site-packages/avocado_vt/plugins/vt_runner.py:230
16:03:48 ERROR| Traceback (most recent call last):
16:03:48 ERROR|   File "/usr/lib/python3.9/site-packages/virttest/error_context.py", line 135, in new_fn
16:03:48 ERROR|     return fn(*args, **kwargs)
16:03:48 ERROR|   File "/usr/lib/python3.9/site-packages/virttest/env_process.py", line 1883, in postprocess
16:03:48 ERROR|     raise RuntimeError("Failures occurred while postprocess:\n%s" % err)
16:03:48 ERROR| RuntimeError: Failures occurred while postprocess:
16:03:48 ERROR| 
16:03:48 ERROR| Postprocess: 'VirtTest' object has no attribute 'outputdir'
```
I think it originates from this line in the env process module:
```python
    if params.get("vm_extra_dump_paths") is not None:
        vm_extra_dumps = os.path.join(test.outputdir, "vm_extra_dumps")
        if not os.path.exists(vm_extra_dumps):
```
Note however that there are other locations in Avocado VT that also make use of this mysterious `outputdir` so later on they should be cleaned up too and potentially set to the same `logdir`. I an do this if you want so feel free to ping me about it once this PR is over.

Reply:
Yes, this is something which should be consolidated if you want to do that I will appreciate your help. Also, we have to keep in our mind that on nrunner those directories are in `/tmp` and anything inside will not be sent to avocado `test-results` directory. That is something which we will have to dealt with on the avocado side.